src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+"""
+Definition of the ``PixelStatisticsCalculator`` class, providing all steps needed to
+calculate the montoring data for the camera calibration.
+"""
+
+import numpy as np
+from astropy.table import Table, vstack
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    ComponentName,
+    Dict,
+    Float,
+    Int,
+    List,
+    TelescopeParameter,
+)
+from ctapipe.monitoring.aggregator import StatisticsAggregator
+from ctapipe.monitoring.outlier import OutlierDetector
+
+__all__ = [
+    "PixelStatisticsCalculator",
+]
+
+
+class PixelStatisticsCalculator(TelescopeComponent):
+    """
+    Component to calculate statistics from calibration events.
+
+    The ``PixelStatisticsCalculator`` is responsible for calculating various statistics from
+    calibration events, such as pedestal and flat-field data. It aggregates statistics,
+    detects outliers, and handles faulty data periods.
+    This class holds two functions to conduct two different passes over the data with and without
+    overlapping aggregation chunks. The first pass is conducted with non-overlapping chunks,
+    while overlapping chunks can be set by the ``chunk_shift`` parameter for the second pass.
+    The second pass over the data is only conducted in regions of trouble with a high percentage
+    of faulty pixels exceeding the threshold ``faulty_pixels_threshold``.
+    """
+
+    stats_aggregator_type = TelescopeParameter(
+        trait=ComponentName(
+            StatisticsAggregator, default_value="SigmaClippingAggregator"
+        ),
+        default_value="SigmaClippingAggregator",
+        help="Name of the StatisticsAggregator subclass to be used.",
+    ).tag(config=True)
+
+    outlier_detector_list = List(
+        trait=Dict(),
```

Comment:
This component is used to aggregate the stats for the _pedestal_, _flatfield_, _timecalibration_, and _pedestalvariance_.  Each of this aggregation has sensible distinct default configuration. Should we prioritize one of the aggregation over the others and set the default config for that aggregation?

Reply:
How do you configure the three separate cases? Our configuration system has the limitation of being able only to configure by class name, not several instances of the same class.

Are these in separate tools? Or are multiple instances of this class created in the same tool?
There are a couple of possibilities depending on that. 

If you only need one instance per application, then leaving the default empty like here is fine.

If you need multiple, differently configured instances of this component, the only way unfortunately is to use inheritance to create several different classes, e.g.:

```python
class FlatFieldStatisticsCalculator(PixelStatisticsCalculator):
    
    @default("outlier_detector_list")
    def default_outlier_config(self):
        return ...
```

Since the plan is to have both types of calibration events in the same stream, I guess you need at least two instances in parallel in the tool.

I think in this case, we can also leave it like this here, and have the two subclasses only inside the tool with the appropriate default configuration.

Reply:
> If you only need one instance per application, then leaving the default empty like here is fine.

Yes, this is what we have foreseen in the workflow. Each calibration type should be processed in a separate stream.